src/shared/models/bank/__tests__/bank.service.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { StakingRepository } from 'src/payment/models/staking/staking.repository';
+import { StakingService } from 'src/payment/models/staking/staking.service';
+import { BuyCryptoService } from 'src/payment/models/buy-crypto/services/buy-crypto.service';
+import { BankAccountService } from 'src/payment/models/bank-account/bank-account.service';
+import { TestSharedModule } from 'src/shared/test.shared.module';
+import { BankAccount } from 'src/payment/models/bank-account/bank-account.entity';
+import { createDefaultBankAccount } from 'src/payment/models/bank-account/__mocks__/bank-account.entity.mock';
+import { BankRepository } from '../bank.repository';
+import { createDefaultBanks } from '../__mocks__/bank.entity.mock';
+
+function createBankSelectorInput(
+  currency: string = 'EUR',
+  amount = 1,
+  bankAccount: BankAccount = createDefaultBankAccount(),
+  kycStatus: KycStatus = KycStatus.COMPLETED,
+): BankSelectorInput {
+  return {
+    bankAccount: bankAccount,
+    amount: amount,
+    currency: currency,
+    kycStatus: kycStatus,
+  };
+}
+
+describe('BankService', () => {
+  let service: BankService;
+
+  let bankRepo: BankRepository;
+  let userService: UserService;
+  let stakingRepo: StakingRepository;
+  let stakingService: StakingService;
+  let buyCryptoService: BuyCryptoService;
+  let fiatService: FiatService;
+  let countryService: CountryService;
+  let bankAccountService: BankAccountService;
+
+  beforeEach(async () => {
+    bankRepo = createMock<BankRepository>();
+    userService = createMock<UserService>();
+    stakingRepo = createMock<StakingRepository>();
+    stakingService = createMock<StakingService>();
+    buyCryptoService = createMock<BuyCryptoService>();
+    fiatService = createMock<FiatService>();
+    countryService = createMock<CountryService>();
+    bankAccountService = createMock<BankAccountService>();
+
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [TestSharedModule],
+      providers: [
+        BankService,
+        { provide: BankRepository, useValue: bankRepo },
+        { provide: UserService, useValue: userService },
+        { provide: StakingRepository, useValue: stakingRepo },
+        { provide: StakingService, useValue: stakingService },
+        { provide: BuyCryptoService, useValue: buyCryptoService },
+        { provide: FiatService, useValue: fiatService },
+        { provide: CountryService, useValue: countryService },
+        { provide: BankAccountService, useValue: bankAccountService },
+        TestUtil.provideConfig(),
+      ],
+    }).compile();
+
+    service = module.get<BankService>(BankService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('should return BF if amount > 9000', async () => {
+    jest.spyOn(countryService, 'getCountryWithSymbol').mockResolvedValue(createDefaultCountry());
+    jest.spyOn(bankRepo, 'find').mockResolvedValue(createDefaultBanks());
```

Comment:
Move the setup methods to a separate setup method (with currency as optional parameter)?